src/chart/rendering/painter/drawline.cpp

```diff
@@ -29,12 +30,7 @@ drawLine::drawLine(const Geom::Line &line,
 	auto wBeg = widths[0] * coordSys.getRect().size.minSize();
 	auto wEnd = widths[1] * coordSys.getRect().size.minSize();
 
-	auto dir = (pEnd - pBeg).normalized();
-
-	auto p0 = pBeg + dir.normal(false) * wBeg;
-	auto p1 = pBeg + dir.normal(true) * wBeg;
-	auto p2 = pEnd + dir.normal(true) * wEnd;
-	auto p3 = pEnd + dir.normal(false) * wEnd;
+	auto trapezoid = ConvexQuad::Isosceles(pBeg, pEnd, wBeg * 2, wEnd * 2);
```

Comment:
Use structure binding : 
auto& [p0, p1, p2, p3] = xpression.points;

Reply:
done. What are the benefits of structure binding here? Just readability?

Reply:
With structure binding the compiler knows which variables want to use, and it will not store the whole structure on the stack and not query members on every usage   (trapezoid.points[0] ... trapezoid.points[1] ...). So probably it can generate more efficient code.